### ISSUE TITLE
General: (_)UNICODE is defined globally while it's only needed by LLVM's IntelJITEvents, which we don't build

### ISSUE THREAD

**Issue by dbrnz** [opencor/opencor#1080]
`cmake/common.cmake` sets UNICODE as a global compile definition. This though breaks Python builds, as `UNICODE` is used as a name in its sources. The global definition only appears to be relevant to LLVM, being used in `LLVM/lib/ExecutionEngine/IntelJITEvents/ittnotify_config.h` which in turn is only included by `LLVM/lib/ExecutionEngine/IntelJITEvents/jitprofiling.c`.

Is there any good reason why the definition of UNICODE (and _UNICODE) cannot be moved into LLVM's CMakeLists.txt file??

**Comment by agarny** [comment]
Good question. You are correct that it's only relevant to LLVM. When I first added support for LLVM in OpenCOR, I was building everything in LLVM and that required defining (_)UNICODE. At the time, the way I was building plugins wasn't as 'advanced' and so I didn't (yet) have a mechanism to specify definitions for a plugin.

Anyway, I have just had a look and it happens that, as I suspected, we don't build `IntelJITEvents` since we not only don't need it (since we use MCJIT instead), but it also requires additional features that may or not be present on a build machine.

In conclusion, it should be safe to remove the definition of (_)UNICODE. (I will do it tomorrow.)
